src/main/java/duke/logic/StorageManager.java

```diff
@@ -0,0 +1,128 @@
+package duke.logic;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.CommonString;
+import duke.task.DeadlineTask;
+import duke.task.DukeTask;
+import duke.task.EventTask;
+import duke.task.TodoTask;
+
+/**
+ * Represents the Storage of Duke.
+ * Manages the storing/loading of data from data file.
+ */
+public class StorageManager {
```

Comment:
Perhaps you can consider splitting this class into multiple classes, such as a Save and Load file which deals with persistent storage and consider having another class for Non-Persistent storage! Making your classes more modular will help you maintain your code in the future!